zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/group/GroupTest.java

```diff
@@ -243,4 +246,40 @@ public void shouldRejectIfEntityIsNotPresentEntityRemoval() {
             "Expected to remove an entity with key '%s' and type '%s' from group with key '%s', but the entity does not exist."
                 .formatted(1L, EntityType.USER, groupKey));
   }
+
+  @Test
+  public void shouldDeleteGroup() {
+    // given
+    final var name = UUID.randomUUID().toString();
+    final var groupKey = engine.group().newGroup(name).create().getValue().getGroupKey();
+    engine
+        .authorization()
+        .permission()
+        .withOwnerKey(groupKey)
+        .withOwnerType(AuthorizationOwnerType.ROLE)
+        .withResourceType(AuthorizationResourceType.ROLE)
+        .withAction(PermissionAction.REMOVE)
+        .add();
+
+    // when
+    final var deletedGroup = engine.group().deleteGroup(groupKey).delete().getValue();
+
+    Assertions.assertThat(deletedGroup)
+        .isNotNull()
+        .hasFieldOrPropertyWithValue("groupKey", groupKey);
```

Comment:
```suggestion
    assertThat(deletedGroup).hasGroupKey(groupKey);
```

🔧 Did you know you can write this a lot easier 😄

Reply:
I did, but the auto-complete kept giving me AssertJ assertions, and I was rushing to wrap up things faster. :D